Ukoly/DU_7/Rodne_cislo.py

```diff
@@ -0,0 +1,101 @@
+MONTH_30 = [4, 6, 9, 11]
+MONTH_31 = [1, 3, 5, 7, 8, 10, 12]
+
+
+def main():
+    number = input('Enter the birth number: ')
+    while len(number) != 11:
+        print('The number has to be 11 characters long')
+        number = input('Enter the birth number: ')
+
+    return number
+
+
+def format(birth_number):
+    print(birth_number)
+    separate = birth_number.split('/')
+    without_slash = ''.join(separate)
+    try:
```

Comment:
Tieto všetky `try` bloky by som zlúčil do jedného, v ktorom sa vykoná všetok kód... a využil by som tam tých správ ktoré vpisuješ do tých výnimiek... Ty keď ju odchytíš pomocou `except` tak sa k tej správe dokážeš dostať, jediné čo musíš spraviť je si prideliť akoby tú samotnú výnimku (jej instanciu -  to čo je instancia si vysvetlíme na OOP) do premennej. To vykonáš nasledujuco: `except ValueError as exc:` (toto spôsobí, že pri odchytení `ValueError` sa ti instancia tej výnimky uloží do premennej `exc`) a následne k tej správe dokážeš pristúpiť pomocou `exc.message` alebo ju natvrdo pretypuješ na string pomocou `str(exc)`. A toto následne môžeš opäť vypísať napr pomocou funkcie `print`, ako to robíš aj teraz.

_Možno taká menšia vsuvka: Toto nie je práve nejlepší spôsob použitia `try` blokov, ale chápem, že na niečom to bolo potrebné precvičiť... Takže možno do budúcna... toto nie je naozaj ako sa to používa :smiley_cat:_

Reply:
Tady jsem to asi nesplnila úplně, sloučila jsem jen dva try bloky do jednoho, protože když int(without_slash) skončilo errorem, vypisovala se defaultní chybová hláška.

Reply:
OK, to znie už celkom dobre :)